### PR TITLE
Use reacji instead of pins

### PR DESCRIPTION
**Pull request by zetlen**
Tired of unpinning things.

Now you can specify an environment variable `REACJI_TO_TRIGGER_TWEET` in your env_variables list. It should be a comma separated list of reacji (e.g. `cow,dash,milky_way,round_pushpin`) that should trigger a tweet.

 - Since that can happen multiple times to the same message, there's a new cache to make sure individual items don't get twote twice.
 - Also refactored the testing a fair amount; it wasn't really doing async correctly.
 - *TODO*: Break out the tweet cache a little better, and add some persistent storage.
 - *TODO*: Add actual tweet IDs to the tweet cache (will require passing them back from the twitter client), and thereby enable untweeting if the last pin reacji is removed?

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -19,25 +20,62 @@ var bot = new Slackbot(require('@slack/client'), env);
...
+  },
+  get: function (item) {
+    return itemCache.get(itemToKey(item));
+  },
+  drop: function (item) {
+    return itemCache.del(itemToKey(item));
+  }
+};
+
+tweetCache.stage({});
```
Comment (jotok) [inline app.js]: Why?

File: app.js [hunk 2]
```diff
@@ -19,25 +20,62 @@ var bot = new Slackbot(require('@slack/client'), env);
 var tweeter = TwitterPoster(env);
 var log = helpers.logger('App', env.isDev);
 
-function postText (message) {
-  console.log(message);
-  var text = message.item.message && message.item.message.text;
+var itemCache = LRU({
```
Comment (jotok) [inline app.js]: It seems like `itemCache` is naturally a member of `tweetCache` and does not need to be in the gobal scope. Is this a js style thing where you don't want to define a class and you don't want to use wacky `(function() { })()` notation?

File: app.js [hunk 3]
```diff
@@ -71,19 +110,35 @@ var postingStrategies = {
   file: postFile
 };
 
-bot.onPinAdded(function (message, channelData) {
+var tweacji = (env.REACJI_TO_TRIGGER_TWEET || 'pushpin').split(',');
+var tweacjiFormatted = tweacji.map(function (ji) { return `:${ji}:`; });
+var tweacjiList = tweacjiFormatted.pop();
+if (tweacjiFormatted.length > 0) {
+  tweacjiList = tweacjiFormatted.join(', ') + ' or ' + tweacjiList;
```
Comment (jotok) [inline app.js]: Is this all to be able to produce a nicely formatted DM?

File: app.js [hunk 4]
```diff
@@ -71,19 +110,35 @@ var postingStrategies = {
...
+}
+
+bot.onItemPinned(function (event, channelData) {
+  var thing = 'something';
+  if (event.item && event.item.message && event.item.message.permalink) {
+    thing = event.item.message.permalink;
+  }
+  var text = `You just pinned ${thing} in #${channelData.channel.name}. Were you trying to get the Pinwheel bot to tweet it? *We've switched to using reacji instead of pins*, because pins are annoying and limited. So go react with ${tweacjiList} instead!`;
```
Comment (jotok) [inline app.js]: Are we planning to keep this forever? Seems like it will get annoying. We should let the user message the bot to acknowledge that they've read the message and then they won't see it again. We can do this in a separate PR (although we should be ready to push a simple PR to remove the DM if people are annoyed by it).

File: app.js (hunk 2 above)
Comment (zetlen) [reply app.js]: Yeah, it's super dirty because it was the last thing I wrote and I was starting to feel sheepish about doing this instead of work. You're exactly right; the public API is `tweetCache` and I oughta break it off into a class.

File: app.js (hunk 1 above)
Comment (zetlen) [reply app.js]: I don't remember?

File: app.js (hunk 3 above)
Comment (zetlen) [reply app.js]: Yeah. I guess I could use some "humanize" library, but when it's four lines of code...

File: app.js (hunk 4 above)
Comment (zetlen) [reply app.js]: Yeah, we definitely want to do that. I actually want to enhance the message with Slack "message buttons", so the user could click one button to unpin-and-reacji instead. But we'd need to have a public callback URL that we could construct from environment variables, like `https://some-wacky-EC2.domain/unpin-and-react/id`.

**Comment by jotok** [comment]
I think this is fine. If you think any of my comments are worth it, we can create issues and/or just fix them in the next PR.
